src/test/java/io/jenkins/pluginhealth/scoring/probes/ProbeEngineTest.java

```diff
@@ -92,4 +95,40 @@ public void shouldNotSaveErrors() {
 
         verify(plugin, never()).addDetails(any());
     }
+
+    @Test
+    public void shouldBeAbleToGetPreviousContextResultInExecution() throws Exception {
+        final Plugin plugin = spy(Plugin.class);
+        final Probe probeOne = new Probe() {
+            @Override
+            protected ProbeResult doApply(Plugin plugin) {
+                return ProbeResult.success(key(), "This is ok");
+            }
+
+            @Override
+            protected String key() {
+                return "foo";
+            }
+        };
+        final Probe probeTwo = new Probe() {
+            @Override
+            protected ProbeResult doApply(Plugin plugin) {
+                return plugin.getDetails().get("foo") != null ?
+                    ProbeResult.success(key(), "This is also ok") :
+                    ProbeResult.error(key(), "This cannot be validated");
+            }
+
+            @Override
+            protected String key() {
+                return "bar";
+            }
+        };
+        final ProbeEngine probeEngine = new ProbeEngine(List.of(probeOne, probeTwo), pluginService);
```

Comment:
In which order of the probes would `probes.forEach` process the 2 probes here?
Because the test would fail if probeTwo gets processed first, I was wondering about how the order is picked up.
Would the test fail if we pass `List.of(probeTwo, probeOne)`?

Reply:
From the `List.of` method implementation, the elements are ordered as they are in parameters. Please look at it.

> Would the test fail if we pass List.of(probeTwo, probeOne)?

Yes. That is how I implemented the `probeTwo`. If it cannot get the probe result from the first probe, it returns an error which are not added into the plugin details. So, we wouldn't have 2 access to `addDetails` method and we wouldn't have  the keys `foo` and `bar` into the details map as key.

Reply:
Thanks for answering in-depth, this is helpful. I read about `List.of` (I should've googled it in the first place) and it helped me to understand this.